R/mod_hierarchical_count_table.R

```diff
@@ -43,135 +217,226 @@ EC <- poc( # nolint
+      log_inform("Partial censor dates failed to convert to date format!", level = "inform")
+  }
+
+  # Prepare event data ----
+
+  event_df2 <- event_df[, c(subjid_var, hierarchy, event_date_var)]
+
+  if (has_event_dt) {
+
+    # NOTE!! Partial dates end up as NA!
+    event_df2[[".event_dt"]] <- as.Date(event_df2[[event_date_var]])
+    if (any(is.na(event_df2[[".event_dt"]]) & !is.na(event_df2[[event_date_var]])))
+      log_inform("Partial event dates failed to convert to date format!", level = "inform")
+
+    # Remove rows with missing event dates
+    event_df2 <- event_df2[!is.na(event_df2[[".event_dt"]]), ]
+  }
+
+  if (has_origin_dt && has_event_dt) {
+
+    # Remove events that occur before origin date (implicitly when origin date is missing)
+    event_df2 <- event_df2 |>
+      dplyr::left_join(pop_df2[, c(subjid_var, ".origin_dt")], by = subjid_var) |>
+      dplyr::filter(.data[[".event_dt"]] >= .data[[".origin_dt"]]) |>
+      dplyr::select(-".origin_dt")
+  }
+
+  if (has_censor_dt && has_event_dt) {
+
+    # Remove events that occur after non-missing censor date
+    event_df2 <- event_df2 |>
+      dplyr::left_join(pop_df2[, c(subjid_var, ".censor_dt")], by = subjid_var) |>
+      dplyr::filter(is.na(.data[[".censor_dt"]]) | .data[[".event_dt"]] <= .data[[".censor_dt"]]) |>
+      dplyr::select(-".censor_dt")
+  }
+
+  if (has_origin_dt && has_censor_dt) {
+
+    # Remove events when origin date is after non-missing censor date (bad data!)
+    event_df2 <- event_df2 |>
+      dplyr::left_join(pop_df2[, c(subjid_var, ".origin_dt", ".censor_dt")], by = subjid_var) |>
+      dplyr::filter(is.na(.data[[".censor_dt"]]) | .data[[".origin_dt"]] <= .data[[".censor_dt"]]) |>
+      dplyr::select(-c(".origin_dt", ".censor_dt"))
+  }
+
+  # Prepare subject level analysis data ----
+
+  adtte <- create_adtte(event_df = event_df2,
+                        pop_df = pop_df2,
+                        hierarchy = hierarchy,
+                        group_var = group_var,
+                        subjid_var = subjid_var,
+                        origin_date_var = ".origin_dt",
+                        censor_date_var = ".censor_dt",
+                        event_date_var = ".event_dt")
+
+  # Drop rows where time at risk could not be determined
+  if (risk) {
+    invalid_rows <- which(is.na(adtte[["AVAL"]]))
+    if (length(invalid_rows) > 0) {
+      invalid_subjects <- unique(adtte[[subjid_var]][invalid_rows])
+      log_inform(paste("Time at risk could not be determined for the following subjects:",
+                       paste(shQuote(invalid_subjects, type = "cmd"), collapse = ", ")), level = "inform")
+      adtte <- adtte[-invalid_rows, ]
+    }
+  }
+
+  # Add group totals ----
+
+  if (total && !is.null(total_group_val) && length(total_group_val) > 0) {
+    adtte_totals <- adtte |>
+      dplyr::mutate(dplyr::across(dplyr::all_of(group_var), ~ total_group_val))
```

Comment:
Would this have the same effect? I think this version should be clearer that `dplyr` in this case.
```suggestion
    adtte_totals <- adtte 
    addtte[[group_var]] <- total_group_val    
```

Reply:
agreed - updated